### ISSUE TITLE
Fast Refresh full reload notification

### ISSUE THREAD

**Issue by Timer** [vercel/next.js#14701]
When Fast Refresh has to fall back to a full page reload, we should first show an informative overlay to the user explaining why Fast Refresh had to bail out. This overlay should include the specific file that caused it, and the link to docs.

Once the user acknowledges this, the full reload should occur. After this, future full page reloads **should not display the overlay**, and instead, reload the page immediately.

This warning should reset every time the dev server is rebooted.

For bonus points, the overlay should also reset once a Fast Refresh is successfully performed in the same session.

**Comment by apollisa** [comment]
Hello there, I would very much like to contribute to Next through this issue!

I took a look at the code and thought about creating a new `BusEvent` specifically for full reload warnings, as well as using `ReactDevOverlay` to display the warning, and to allow the user to refresh.

The `BusEvent` would be dispatched in `hot-dev-client.js` instead of the `console.warn`.

As for displaying the warning only once, I thought about using the local storage, much the same way it was suggested in #13588.

Does that seem reasonable to you?

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
